Yosemite/Yosemite/PointOfSale/PointOfSaleCouponService.swift

```diff
@@ -51,31 +51,37 @@ public final class PointOfSaleCouponService: PointOfSaleCouponServiceProtocol {
                   storage: storage)
     }
 
-    // Provides an array of coupons that are stored locally, it does not accept any sort of pagination
+    // Provides an array of coupons that are stored locally
+    // It does not accept any sort of pagination
+    // Limited to default page size to match remote results
     @MainActor
     public func provideLocalPointOfSaleCoupons() async throws -> [POSItem] {
+        return try await provideLocalPointOfSaleCoupons(limit: Constants.defaultPageSize)
+    }
+
+    @MainActor
+    private func provideLocalPointOfSaleCoupons(limit: Int?) async throws -> [POSItem] {
         let couponsEnabled = try await checkStoreCouponSettings()
         if !couponsEnabled {
             throw PointOfSaleCouponServiceError.couponsDisabled
         }
 
-        let localCoupons = await fetchLocalCoupons()
-        return localCoupons.items
+        return fetchLocalCoupons(limit: limit)
     }
 
     @MainActor
     public func providePointOfSaleCoupons(pageNumber: Int) async throws -> PagedItems<POSItem> {
         do {
-            try await syncCouponsFromRemote(pageNumber: pageNumber)
+            let hasMorePages = try await syncCouponsFromRemote(pageNumber: pageNumber)
+            let coupons = try await provideLocalPointOfSaleCoupons(limit: nil)
```

Comment:
Perhaps this could use some comments to explain exactly what we do? It's a little confusing to sync from the remote, then provide the local coupons if that succeeds.

In the context of this PR I can understand it but I can see people getting thrown by it in future.